### PR TITLE
#25290 Fixing Internal QA

### LINKED ISSUE #25290: Build New Search Engine Results Page Preview Tool for Google
### Parent Issue

_No response_

### Task

Build New Search Engine Results Page Preview Tool for Google

### Proposed Objective

Same as Parent Issue

### Proposed Priority

Same as Parent Issue

### Acceptance Criteria

See notes here:

* Social Media Tile & Search Engine Results Page Preview Tool Messages | dotCMS - Google Docs || https://docs.google.com/document/d/1O8PMni7FzD9YmIHdz-De43MnHpO--0_W49ZwYxLt9DM/edit

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

Demo - https://www.figma.com/proto/jyu7b1wgL16iV7yOoNdgu9/SEO?page-id=1%3A65&type=design&node-id=298-6800&viewport=-1440%2C-118%2C0.37&scaling=scale-down&starting-point-node-id=372%3A9190 

Design - https://www.figma.com/file/jyu7b1wgL16iV7yOoNdgu9/SEO?type=design&node-id=523-11475&t=URNtP1TkxmYhhpH9-0

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by manuelrojas**
### Summary
🖥️🛠️🎨

Added a device selector feature to the SEO tab of the edit page state controller. This feature allows users to preview the page SEO data on different devices and toggle the overlay panel with an icon button. Refactored some code to improve readability, conciseness, and testability. Fixed a minor styling issue with the dot tab buttons component.

> _Sing, O Muse, of the glorious deeds of the `DotDeviceSelectorSeoComponent`,_
> _Which lets the wise and skillful editors preview the page on different screens,_
> _And with a click of the `dot-tab-dropdown__icon`, adorned with various shapes,_
> _They can hide or show the overlay panel, like Zeus conceals or reveals the clouds._

### Walkthrough
*  Add `(onHide)` event binding to `dot-device-selector-seo` component to handle overlay panel hiding ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-20ee3da83820dff9c0690885a51f7e6740749b9a33bcf6469f67dfaa43463672R3))
*  Add `@Output() hideOverlayPanel` property to `DotDeviceSelectorSeoComponent` class to emit event when overlay panel is hidden ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-764ca6ad6e5980c9f3ddfceee590e38984f85ed0ac63dba184a37e2ef955c5c4R53))
*  Add `onHideDeviceSelector` and `openMenu` methods to `DotDeviceSelectorSeoComponent` class to handle overlay panel hiding and showing and icon changing ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-764ca6ad6e5980c9f3ddfceee590e38984f85ed0ac63dba184a37e2ef955c5c4R186-R192))
*  Add `(hideOverlayPanel)` event binding to `dot-edit-page-state-controller-seo` component to handle overlay panel hiding from parent component ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-c33fbce2e3590bf75f68ecc21dbdd226cf8301b16259e8029d4d6f426f9e75d4R7))
*  Add `[dotTabsIcon]` input binding and `(openMenu)` event binding to `dot-tab-buttons` component to pass icon and event from parent component ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-c33fbce2e3590bf75f68ecc21dbdd226cf8301b16259e8029d4d6f426f9e75d4L14-R16))
*  Add `@ViewChild('deviceSelector') deviceSelector` property to `DotEditPageStateControllerSeoComponent` class to access child component ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-525d72b2f066845092b3f8e47374c999304f59f8d049c44bb98ff91fddcac347R76))
*  Add `dotTabsIcon` property to `DotEditPageStateControllerSeoComponent` class to store icon for tab buttons component ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-525d72b2f066845092b3f8e47374c999304f59f8d049c44bb98ff91fddcac347R82))
*  Add `onHideDeviceSelector` and `openDeviceSelector` methods to `DotEditPageStateControllerSeoComponent` class to handle overlay panel hiding and showing and icon changing from parent component ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-525d72b2f066845092b3f8e47374c999304f59f8d049c44bb98ff91fddcac347R200-R217))
*  Remove redundant assignment of `this.currentResults = this.seoOGTagsResults` from `DotResultsSeoToolComponent` class ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-6084f02c375b9edee18f429040e76a983a8469637f28dd29a9c8fa330774fb88L83))
*  Move `dot-tab-dropdown__icon` class from `i` element to parent `button` element and use `[class]="dotTabsIcon"` binding instead of `[class]` binding in `dot-tab-buttons` component template to fix styling issue and use input property ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-1b160a493f0ac718710f1d6617775800e76011808545a2c2d6b9c7bbb1676ae9L23-R27))
*  Add `data-testId` attribute to `i` element in `dot-tab-buttons` component template for testing purposes ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-1b160a493f0ac718710f1d6617775800e76011808545a2c2d6b9c7bbb1676ae9L23-R27))
*  Add `dotTabsIcon` input property to `DotTabButtonsComponent` spectator initialization to pass icon value ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-103bc995132680fbeaa0e1e9900a7b0d8918456776184e6d67884a4d7c1f1584L37-R38))
*  Update test for `toggleIcon` method to use `dotTabsIcon` input property and `byTestId` selector in `DotTabButtonsComponent` spec file ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-103bc995132680fbeaa0e1e9900a7b0d8918456776184e6d67884a4d7c1f1584L93-R108))
*  Add `dotTabsIcon` input property and remove `dropDownOpenIcon`, `dropDownCloseIcon`, and `icon` properties and `toggleIcon` method from `DotTabButtonsComponent` class as they are no longer needed ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-bf8bca39c1833ed819e57a1eb3902053bef01922de6f904b73c5a7e4ad6a5e53L30-R34), [link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-bf8bca39c1833ed819e57a1eb3902053bef01922de6f904b73c5a7e4ad6a5e53L55-R55))
*  Improve comment for `getFilteredMetaTagsByMedia` method in `dot-seo-meta-tags.service.ts` to specify filtering criteria ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-e2c6edc96a3d4a7581030dec6768557e11cfc76d6863da2ff346f2cb3efbc715L137-R141))
*  Simplify conditions for checking `ogDescription` using optional chaining operator in `dot-seo-meta-tags.service.ts` ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-e2c6edc96a3d4a7581030dec6768557e11cfc76d6863da2ff346f2cb3efbc715L224-R223), [link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-e2c6edc96a3d4a7581030dec6768557e11cfc76d6863da2ff346f2cb3efbc715L232-R231))
*  Add comment for `isCMSAdmin` method in `dot-device-selector-seo.component.ts` to describe its purpose ([link](https://github.com/dotCMS/core/pull/26067/files?diff=unified&w=0#diff-764ca6ad6e5980c9f3ddfceee590e38984f85ed0ac63dba184a37e2ef955c5c4R174-R177))

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots

https://github.com/dotCMS/core/assets/3438705/83f121ef-5ffd-4f90-b59b-62afc54545f0

### CODE REVIEW

**Review by zJaaal — COMMENTED** [review]
I think I made the dropdown change the icon when you click outside the panel/button by listening to the document click, see my PR I made several changes in `dot-tab-buttons` https://github.com/dotCMS/core/pull/26032

**Review by hmoreras — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Comment by manuelrojas** [comment]
> I think I made the dropdown change the icon when you click outside the panel/button by listening to the document click, see my PR I made several changes in `dot-tab-buttons` #26032

We are going to merge this one and then fix the merge conflicts about this @zJaaal @fmontes

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.ts [hunk 1]
```diff
@@ -178,6 +183,13 @@ export class DotDeviceSelectorSeoComponent implements OnInit {
         );
     }
 
+    /**
+     * Hide the overlay panel
+     */
+    onHideDeviceSelector() {
+        this.hideOverlayPanel.emit();
+    }
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.ts:191]: This need a test.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-state-controller-seo/dot-edit-page-state-controller-seo.component.html [hunk 2]
```diff
@@ -4,14 +4,16 @@
...
+    (hideOverlayPanel)="onHideDeviceSelector()"
     appendTo="body"
     tooltipPosition="bottom"
     tooltipStyleClass="dot-device-selector__dialog"
     data-testId="dot-device-selector"></dot-device-selector-seo>
 <dot-tab-buttons
     [mode]="mode"
     [options]="options"
-    (openMenu)="deviceSelector.openMenu($event)"
+    [dotTabsIcon]="dotTabsIcon"
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-state-controller-seo/dot-edit-page-state-controller-seo.component.html:15]:
```suggestion
    [icon]="dotTabsIcon"
```

File: core-web/libs/ui/src/lib/dot-tab-buttons/dot-tab-buttons.component.ts [hunk 3]
```diff
@@ -27,12 +27,11 @@ export class DotTabButtonsComponent {
     @Output() clickOption = new EventEmitter();
```
Comment (fmontes) [inline core-web/libs/ui/src/lib/dot-tab-buttons/dot-tab-buttons.component.ts:27]: Adding the click to a div have a11y issues, why don't put the click directly to the buttons?

Comment (fmontes) [reply core-web/libs/ui/src/lib/dot-tab-buttons/dot-tab-buttons.component.ts:27]: Also... if the parent div should be the host, this: https://dotcms.slack.com/archives/C01VC00SA5A/p1694626261079379

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-state-controller-seo/dot-edit-page-state-controller-seo.component.html [hunk 4]
```diff
@@ -4,14 +4,16 @@
...
+    (openMenu)="openDeviceSelector($event)"
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-state-controller-seo/dot-edit-page-state-controller-seo.component.html:16]:
```suggestion
    (dropdownClick)="openDeviceSelector($event)"
```

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Pull request merged**